src/server/utils/tx-manager/queueMongo.js

```diff
@@ -244,11 +219,10 @@ export default class queueMongo {
   async run() {
     try {
       if (this.queue.length > 0) {
-        const nextTr = this.queue[0]
+        const nextTr = this.queue.shift()
 
         const walletNonce = await this.getWalletNonce(nextTr.addresses)
         if (walletNonce) {
-          this.queue.shift()
```

Comment:
What if there's an exception in the above await? you need to restore the TX to the queue

Reply:
it was the main problem why it is not working with many wallets... ok  I will add try catch which return it to queue